src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -50,6 +55,22 @@ public static Name parseName(String name) throws ParseException {
         return new Name(trimmedName);
     }
 
+    /**
+     * Parses a {@code String name} into a {@code Name}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code name} is invalid.
+     */
+    public static seedu.address.model.income.Name parseIncomeName(String name) throws ParseException {
+        requireNonNull(name);
+        String trimmedName = name.trim();
+        if (!Name.isValidName(trimmedName)) {
+            throw new ParseException(Name.MESSAGE_CONSTRAINTS);
+        }
+        return new seedu.address.model.income.Name(trimmedName);
```

Comment:
Yes, as of now I just left it as this first since I didn't want to delete any files relating to Person. But it's still something we have to discuss even when we add Expense since it has a Name field as well